gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -416,15 +417,22 @@ func (s *server) gcsRequest(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// e.g. "/gcs/bucket/path/to/object" -> "/bucket/path/to/object"
-	path := strings.TrimPrefix(r.URL.Path, gcsPath)
-	// e.g. "/bucket/path/to/object" -> ["bucket", "path/to/object"]
-	bucket, object := splitBucketObject(path)
-	objectLogger := logger.WithFields(logrus.Fields{"bucket": bucket, "object": object})
+	path := r.URL.Path
+	prowPath, err := parsePath(path)
+	if err != nil {
+		logger.WithError(err).Error("error parsing path")
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "Error: %v", err)
+		return
+	}
+
+	// e.g. "/gcs/bucket/path/to/object/" -> "path/to/object"
+	object := strings.Trim(prowPath.Path, "/")
```

Comment:
It doesn't seem like this does what the comment above it says it does any longer